### DISCUSSION TITLE
Issue with scale bar unit

### DISCUSSION THREAD

**Discussion by Bhowmik1995** [mtex-toolbox/mtex#1668] [Ask Anything]
![image](https://user-images.githubusercontent.com/119377783/204417061-17e39e2f-0c17-48aa-bb84-78c22617ab52.png)

Hi, 
I have been trying to plot with a scale bar in a micrometer unit. But I am only getting the scale bar unit as a nanometer. Here is the code I am using to the following lines to modify the scalebar parameter:

f=gcm;
mp=getappdata(f.currentAxes,'mapPlot');
mp.micronBar.scanUnit = 'um';
mp.micronBar.txt.FontSize = 16; % change font size
mp.micronBar.length = scale_bar; % change length

Can anyone please help to solve this issue?

Shubhro

**Comment by hakonanes** [comment]
You can set the length and then overwrite the `micronBar.txt.String`, however you should be very careful to set the correct text.

In my case I have a scalebar showing "500 nm" while I want "1 um" for the scalebar to be in line with plots made with other software. Updating `micronBar.scanUnit = 'um'` and setting length to 1 do not work. This works:

```matlab
figure
[~, mP] = plot(ebsd, ebsd.some_property)
mP.micronBar.length = 1000;
pause(0.1) % Was necessary so that MATLAB/MTEX finished updating the figure before my update
mP.micronBar.txt.String = '\rm{\textbf{1 $\mu$m}}';
```
